src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -248,17 +263,44 @@ private static boolean shouldBeOnSameLine(RightCurlyOption bracePolicy, Details
      * @return true if a right curly should be alone on a line.
      */
     private static boolean shouldBeAloneOnLine(RightCurlyOption bracePolicy, Details details) {
+        final int tokenLambda = details.lcurly.getParent().getType();
         return bracePolicy == RightCurlyOption.ALONE
+                && tokenLambda != TokenTypes.LAMBDA
                 && !isAloneOnLine(details)
                 && !isEmptyBody(details.lcurly)
-                || bracePolicy == RightCurlyOption.ALONE_OR_SINGLELINE
+                || shouldBeAloneInLambda(bracePolicy, details)
+                || shouldBeAloneOrSingleLine(bracePolicy, details);
+    }
```

Comment:
lets try.
but name method as "isValidAloneOnLine" or different way but without "General". code should be read as text.